test/spec/spectest.cpp

```diff
@@ -465,55 +466,61 @@ bool SpecTest::stringContains(std::string_view Expected,
 
 void SpecTest::run(std::string_view Proposal, std::string_view UnitName) {
   spdlog::info("{} {}", Proposal, UnitName);
-  std::ifstream JSONIS(TestsuiteRoot / Proposal / UnitName /
-                       (std::string(UnitName) + ".json"s));
-  rapidjson::IStreamWrapper JSONISWrapper(JSONIS);
-  rapidjson::Document Doc;
-  auto &Allocator = Doc.GetAllocator();
-  Doc.ParseStream(JSONISWrapper);
+  auto FName =
+      (TestsuiteRoot / Proposal / UnitName / (std::string(UnitName) + ".json"s))
+          .string();
+
+  simdjson::dom::parser parser;
+  simdjson::dom::element Doc;
+  try {
+    Doc = parser.load(FName);
+  } catch (simdjson::error_code &err) {
+    ;
+  }
 
   std::map<std::string, std::string> Alias;
   std::string LastModName;
 
   // Helper function to get module name.
-  auto GetModuleName = [&](const rapidjson::Value &Action) -> std::string {
-    if (const auto &Module = Action.FindMember("module"s);
-        Module != Action.MemberEnd()) {
-      // Get the module name.
-      auto ModName = Module->value.Get<std::string>();
-      if (auto It = Alias.find(ModName); It != Alias.end()) {
+  auto GetModuleName = [&](const simdjson::dom::object &Action) -> std::string {
+    std::string_view ModName;
+    if (!Action["module"].get(ModName)) {
+      if (auto It = Alias.find(std::string(ModName)); It != Alias.end()) {
         // If module name is aliased, use the aliased name.
         return It->second;
       }
-      return ModName;
+      return std::string(ModName);
     }
     return LastModName;
   };
 
-  auto Invoke = [&](const rapidjson::Value &Action,
-                    const rapidjson::Value &Expected, uint64_t LineNumber) {
+  auto Invoke = [&](const simdjson::dom::object &Action,
+                    const simdjson::dom::array &Expected, uint64_t LineNumber) {
     const auto ModName = GetModuleName(Action);
-    const auto Field = Action["field"s].Get<std::string>();
-    const auto Params = parseValueList(Action["args"s]);
+    const std::string_view Field = Action["field"];
+    simdjson::dom::array Args = Action["args"];
```

Comment:
Do we need this temp variable? Is it possible to use `parseValueList(Action["args"])`.

Reply:
The problem is by default, when indexing the parsed objects, SIMDJSON returns `simdjson::result<simdjson::dom::value>`, which doesn't convert to `simdjson::dom::array` implicitly
 Hence, I had to use this temp variable, to type cast it explicitly.

Reply:
Does SIMDJSON provide `get_array` or something? This sounds so weird.

Reply:
It does, but then it's still gives `simdjson::result<simdjson::dom::array>`, which still needs to be converted